rusoto/credential/src/lib.rs

```diff
@@ -383,23 +396,31 @@ impl ChainProvider {
 async fn chain_provider_credentials(
     provider: ChainProvider,
 ) -> Result<AwsCredentials, CredentialsError> {
-    if let Ok(creds) = provider.environment_provider.credentials().await {
-        return Ok(creds);
+    let mut err = CredentialsError {
+        message: "".to_owned(),
+    };
```

Comment:
```suggestion
    let mut err = CredentialsError::new("Couldn't find AWS credentials in environment, credentials file, or IAM role.");
```

and remove the last `.merge_with_str` at line 420